### ISSUE TITLE
Thoughts from the TD Use Case Discussions

### ISSUE THREAD

**Issue by egekorkan** [w3c/wot-usecases#261]
In the previous TD meetings, we had some discussions about what we need from use cases process. These are currently documented in [the meeting agenda](https://www.w3.org/WoT/IG/wiki/WG_WoT_Thing_Description_WebConf#Use_Case_Discussion) but I would like to move them here. I am copying them below to give them a place. We can discuss this in a use case call:

Overall thoughts:

* Use Cases should be iterated to extract the information we need.
  * Relates to https://github.com/w3c/wot-usecases/issues/192
* New use cases should at least have the information we need as a TF. We can filter out current use cases if they cannot fulfill the new template requirements, thus at least the TD TF requirements.
* We can extract User Stories from Use Cases. A User Story contains what the user (developer) wants to do with WoT (TDs) and whether are they able to do it. E.g. as User X, I want to ABC so I can achieve XYZ. It is better if the user or the persona is visible in the User Story.
  * The TD TF thinks that if they are submitting a User Story, they are not able to do what they want to do.

Backtracking some additions in TD 1.1 and 1.0:

* Thing Models
  * Use Case: Managing/Representing a set of instantiations of the same model of a device. Correlating (search) in a collection of TDs that represent the same model of the device. 
  * User Story: As a device manufacturer, I want to write a TD-like document only once. TDs represent an instance but I need to represent the generic model of that device that each instance refers to via their TDs. If this is not standardized, I cannot guarantee that my customers understand this generic model representation.
  * Requirement: One file to represent a model of the device
  * Feature: Thing Model mechanism
* Extension/Submodel in Thing Models
  * Use Case: Reusability. Consistent affordance definition for reuse, e.g. of temperature.
  * User Story: As a device manufacturer, I want to reuse existing definitions and vocabularies in Thing Models. Semantic Ontologies are not enough since they may not result in concrete JSON serialization in a TM or TD.
  * Requirement: Reuse of existing definitions found in Thing Models
  * Feature: tm:submodel, tm:extends etc.
* synchronous word in actions
  * Use Case: Interacting with two robots that behave slightly differently (one replying right away and moving, the other moving and then replying)
  * User Story: As a Consumer application programmer, I want to do less trial and error when programming a new Consumer application. However, invoking an action can return a response right away or after the physical process has finished. Not knowing the difference in advance results in a trial-and-error process.
  * Requirement: A keyword indicating this behavior
  * Feature: "synchronous" keyword with boolean type

Note: I can add more backtracking examples.

**Comment by mmccool** [comment]
A discussion in another issue that I think belongs here: https://github.com/w3c/wot-usecases/issues/257#issuecomment-1907308039

**Comment by mmccool** [comment]
Related to discussion in meeting Jan 24: sometimes the user story comes first, but it still needs to be related to one (or more) use cases.   To take an example above, the "synchronous word in actions" relates to robots, which are used in Factory automation.  But then it talks about "Consumer applications", but probably this should be "Factory automation engineer"?  Anyway, thinking about the use case can clarify the user story :)

**Comment by mmccool** [comment]
Also related to point of how to get "detailed" use cases: we could ask submitters to provide a use case (application domain example) and then also submit a set of user stories.   If there aren't any user stories motivated by a use case then there are no gaps and...

**Comment by chachamimm** [comment]
Features that we want to implement should be based on needs (requirements) from industry. So we have to collect a lot of use cases from industry and analysis them and get needs (requirements). If you can get the needs from user stories, I think user stories are also important.
